# ans to question no 1

In [1]:
import numpy as np

In [2]:
my_array=np.zeros(10)
print(my_array)

[0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]


# ans to question no 2

In [3]:
my_array[4]=1
print(my_array)

[0. 0. 0. 0. 1. 0. 0. 0. 0. 0.]


# ans to question no 3

In [4]:
arr2=np.array([(0,1,2),(3,4,5),(6,7,8)])

In [5]:
print(arr2)

[[0 1 2]
 [3 4 5]
 [6 7 8]]


# ans to question no 4

In [6]:
import random
arr3=np.zeros((3,3,3))
string='0123456789'
for i in range(0,3):
    
    for j in range(0,3):
        for k in range(0,3):
            arr3[i,j,k]=random.choice(string)
print(arr3)            
    
    

[[[5. 8. 2.]
  [0. 4. 8.]
  [1. 2. 2.]]

 [[1. 8. 2.]
  [8. 2. 2.]
  [1. 3. 8.]]

 [[0. 9. 1.]
  [2. 8. 5.]
  [6. 0. 4.]]]


# ans to question no 5

In [7]:
arr1=np.zeros((5,3))
for i in range(0,5):
    for j in range(0,3):
        arr1[i,j]=random.choice(string)
arr2=np.zeros((3,2))
for i in range(0,3):
    for j in range(0,2):
        arr2[i,j]=random.choice(string)
arr3=np.dot(arr1,arr2)
print(arr1)
print(arr2)
print(arr3)

[[7. 8. 2.]
 [3. 2. 0.]
 [7. 0. 1.]
 [0. 6. 3.]
 [3. 3. 9.]]
[[5. 3.]
 [9. 6.]
 [7. 7.]]
[[121.  83.]
 [ 33.  21.]
 [ 42.  28.]
 [ 75.  57.]
 [105.  90.]]


# ans to question no 6

In [8]:
arr=np.zeros((5,5))
for i in range(0,5):
    for j in range(0,5):
        arr[i,j]=j
print(arr)        

[[0. 1. 2. 3. 4.]
 [0. 1. 2. 3. 4.]
 [0. 1. 2. 3. 4.]
 [0. 1. 2. 3. 4.]
 [0. 1. 2. 3. 4.]]


# ans to question no 7

In [9]:
print(arr1)

[[7. 8. 2.]
 [3. 2. 0.]
 [7. 0. 1.]
 [0. 6. 3.]
 [3. 3. 9.]]


In [10]:
lst=[]
for i in range(0,5):
    lst.append(arr1.mean(axis=1)[i])
for i in range(0,5):
    for j in range(0,3):
        arr1[i,j]=arr1[i,j]-lst[i]
print(arr1)        
    

[[ 1.33333333  2.33333333 -3.66666667]
 [ 1.33333333  0.33333333 -1.66666667]
 [ 4.33333333 -2.66666667 -1.66666667]
 [-3.          3.          0.        ]
 [-2.         -2.          4.        ]]


# ans to question no 8

In [11]:
# we calculated the dot product in equation no 3
print(arr3)

[[121.  83.]
 [ 33.  21.]
 [ 42.  28.]
 [ 75.  57.]
 [105.  90.]]


In [12]:
for i in range(0,5):
    for j in range(0,2):
        if(i==j):
            print(arr3[i,j])

121.0
21.0


# ans to question no 9

In [13]:
import random
arr3=np.zeros((3,3,3))
string='0123456789'
for i in range(0,3):
    
    for j in range(0,3):
        for k in range(0,3):
            arr3[i,j,k]=random.choice(string)
print(arr3)            
    


[[[7. 3. 7.]
  [2. 1. 7.]
  [7. 0. 3.]]

 [[2. 7. 9.]
  [9. 3. 9.]
  [5. 8. 2.]]

 [[6. 7. 6.]
  [8. 9. 2.]
  [0. 1. 8.]]]


In [14]:
from statistics import mode
lst=[]


for i in range(0,3):
    for j in range(0,3):
        for k in range(0,3):
            t=arr3[i,j,k]
            lst.append(t)
        
print('the most frequent value is',mode(lst))        

the most frequent value is 7.0


# ans to question no 10

In [15]:
random_vector=np.random.random(10)
print(random_vector)# it gives the random vector 
random_vector.sort()# it sorts the vector within the array and does not return anything
print(random_vector)# now we print the random vector to see if the changes have been made

[0.27420081 0.79956836 0.94509726 0.79815232 0.52432462 0.33106653
 0.34915558 0.17380837 0.86205591 0.32574984]
[0.17380837 0.27420081 0.32574984 0.33106653 0.34915558 0.52432462
 0.79815232 0.79956836 0.86205591 0.94509726]


# question of pandas

# ans to question no 1

In [16]:
import pandas as pd
from numpy import genfromtxt
import csv

In [19]:
chipo=pd.read_csv('../datasets/chipotle.tsv',sep='\t')
print(chipo)

      order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
5            3         1                           Chicken Bowl   
6            3         1                          Side of Chips   
7            4         1                          Steak Burrito   
8            4         1                       Steak Soft Tacos   
9            5         1                          Steak Burrito   
10           5         1                    Chips and Guacamole   
11           6         1                   Chicken Crispy Tacos   
12           6         1                     Chicken Soft Tacos   
13           7         1                           Chicken Bow

## ans to question no 2

In [20]:
chipo.head(10)


order_id  quantity                              item_name  \
0         1         1           Chips and Fresh Tomato Salsa   
1         1         1                                   Izze   
2         1         1                       Nantucket Nectar   
3         1         1  Chips and Tomatillo-Green Chili Salsa   
4         2         2                           Chicken Bowl   
5         3         1                           Chicken Bowl   
6         3         1                          Side of Chips   
7         4         1                          Steak Burrito   
8         4         1                       Steak Soft Tacos   
9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                                NaN     $2.39   
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN     $2.39   
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                NaN     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

# ans to question no 3

In [21]:
chipo.count()

order_id              4622
quantity              4622
item_name             4622
choice_description    3376
item_price            4622
dtype: int64

it gives me the total no no of observations in the dataset which is equal to 4622

# ans to question no 5

In [22]:
chipo.columns

Index(['order_id', 'quantity', 'item_name', 'choice_description',
       'item_price'],
      dtype='object')

# ans to question no 4

In [23]:
len(chipo.columns)

5

this is how we calculate the no of columns ina dataset

# ans to question no 7 and 8

In [24]:
df=chipo.groupby('item_name')['quantity']
n_dt=df.count()
item = max (n_dt)

for i in range (len (n_d)):
    t=df[i]
    if t==item:
        print (st,'is the most ordered item')
        break
        

NameError: name 'n_d' is not defined

In [26]:
print(df)

chicken bowl was most ordered

In [ ]:
df=chipo.groupby('item_name')['quantity'].count()

In [ ]:
dir(df)

In [ ]:
df.iloc[:]

## ans to question no 6

In [ ]:
chipo.index

this is how the data set has been indexed starting from zero ending at 4622 and having a step value of 1

# ans to question no 10

In [ ]:
chipo['quantity'].sum()

thus a total of 4972 items were ordered

# ans to question no 11(to turn item price into float)

In [ ]:
chipo['item_price'].dtype

clearly it is a string value

In [ ]:
lst=[]
for i in range(len(chipo)):
    string=chipo['item_price'][i].split('$')[1]
    lst.append(float(string))
    

In [ ]:
chipo['price_float']=lst

In [ ]:
chipo.head()

clearly the prices have been displayed in float values in a new column

# ans to question no 12

In [ ]:
chipo['price_float'].dtype

clearly the type has been changed to float

# ans to question no 13

In [ ]:
chipo['price_float'].sum()

the revenue for the period was 34500 dollars

## ans to question no 14

In [ ]:


len(chipo['item_name'].unique())

thus a total of 50 unique items were sold

# ans to question no 15

In [ ]:
length=len(chipo['order_id'].unique())
print(length)

thus a total of 1834 orders were made during the time

# ans to question no 16

In [ ]:
sum(chipo['price_float'])/length